### PR TITLE
Feature/fetch task from gelato

### PR DESCRIPTION
**Pull request by superarius**
New feature for omen-subgraph to track tasks on Gelato.

EDIT: we had some issues compiling on different setups (dependency versioning issues) but we eventually resolved the issues and have upgraded from draft PR to "ready for review"

### CODE REVIEW

**Review by KadenZipfel — APPROVED** [review]

File: package.json [hunk 1]
```diff
@@ -44,13 +44,13 @@
     "@gnosis.pm/conditional-tokens-contracts": "^1.0.1",
     "@gnosis.pm/conditional-tokens-market-makers": "^1.8.1",
     "@gnosis.pm/truffle-nice-tools": "^1.3.1",
-    "@graphprotocol/graph-cli": "^0.18.0",
-    "@graphprotocol/graph-ts": "^0.18.1",
+    "@graphprotocol/graph-cli": "^0.19.0",
+    "@graphprotocol/graph-ts": "^0.19.0",
```
Comment (cag) [inline package.json:48]: Oh interesting, I made a conflict by merging something earlier, oops...

Should be an easy fix. I actually prefer the way it is written in this PR.

File: package.json [hunk 2]
```diff
@@ -63,7 +63,7 @@
     "realitio-gnosis-proxy": "github:gnosis/realitio-gnosis-proxy",
     "replace-in-file": "^6.1.0",
     "should": "^13.2.3",
-    "truffle": "^5.1.18",
+    "truffle": "5.1.35",
```
Comment (cag) [inline package.json:66]: Interesting, should Truffle be version pinned?

File: schema.graphql [hunk 3]
```diff
@@ -261,6 +261,82 @@ type FpmmTransaction @entity {
   transactionHash: Bytes!
 }
 
+type User @entity {
```
Comment (cag) [inline schema.graphql:264]: Rename to GelatoUser?

File: subgraph.template.yaml [hunk 4]
```diff
@@ -34,6 +34,35 @@ dataSources:
...
+    network: {{network}}
+    source:
+      address: '{{GelatoCore.address}}'
+      abi: GelatoCore
+      startBlock: {{GelatoCore.startBlock}}
+    mapping:
+      kind: ethereum/events
+      apiVersion: 0.0.3
+      language: wasm/assemblyscript
+      entities:
```
Comment (cag) [inline subgraph.template.yaml:48]: Entities are the `@entity` definitions in the schema.graphql, not events or handlers.

**Review by cag — COMMENTED** [review]
I'm kind of confused actually. This part of the subgraph looks completely self-contained. Couldn't it just exist as a separate Gelato subgraph? What's the value in bundling this code along with the Omen subgraph?

**Comment by hilmarx** [comment]
>I'm kind of confused actually. This part of the subgraph looks completely self-contained. Couldn't it just exist as a separate Gelato subgraph? What's the value in bundling this code along with the Omen subgraph?

@cag so that is actually what we started with and we also have isolated Gelato subgraphs ready and deployed on [Mainnet](https://thegraph.com/explorer/subgraph/gelatodigital/gelato-network) & [Rinkeby](https://thegraph.com/explorer/subgraph/gelatodigital/gelato-network-rinkeby), however @pimato mentioned that for the Omen-Exchange UI integration it would be better to have only one subgraph that contains every query for all the various underlying protocols, as adding multiple different ones using Apollo can create added complexity. Thus we made this PR. 

I think from a UI integrations perspective @pimato is right about that. Also we will submit a PR to the omen-exchange repo soon that will have a functioning integration based on this commit of the subgraph. 

@pimato your take?

File: package.json (hunk 2 above)
Comment (hilmarx) [reply package.json:66]: nope, fixed to `^5.1.35`

File: schema.graphql (hunk 3 above)
Comment (hilmarx) [reply schema.graphql:264]: renamed to `GelatoUser` to make things clearer.

File: subgraph.template.yaml (hunk 4 above)
Comment (hilmarx) [reply subgraph.template.yaml:48]: fixed.

File: package.json (hunk 1 above)
Comment (hilmarx) [reply package.json:48]: fixed `d8c9763`

**Comment by pimato** [comment]
> > I'm kind of confused actually. This part of the subgraph looks completely self-contained. Couldn't it just exist as a separate Gelato subgraph? What's the value in bundling this code along with the Omen subgraph?
> 
> @cag so that is actually what we started with and we also have isolated Gelato subgraphs ready and deployed on [Mainnet](https://thegraph.com/explorer/subgraph/gelatodigital/gelato-network) & [Rinkeby](https://thegraph.com/explorer/subgraph/gelatodigital/gelato-network-rinkeby), however @pimato mentioned that for the Omen-Exchange UI integration it would be better to have only one subgraph that contains every query for all the various underlying protocols, as adding multiple different ones using Apollo can create added complexity. Thus we made this PR.
> 
> I think from a UI integrations perspective @pimato is right about that. Also we will submit a PR to the omen-exchange repo soon that will have a functioning integration based on this commit of the subgraph.
> 
> @pimato your take?

You are right

> > I'm kind of confused actually. This part of the subgraph looks completely self-contained. Couldn't it just exist as a separate Gelato subgraph? What's the value in bundling this code along with the Omen subgraph?
> 
> @cag so that is actually what we started with and we also have isolated Gelato subgraphs ready and deployed on [Mainnet](https://thegraph.com/explorer/subgraph/gelatodigital/gelato-network) & [Rinkeby](https://thegraph.com/explorer/subgraph/gelatodigital/gelato-network-rinkeby), however @pimato mentioned that for the Omen-Exchange UI integration it would be better to have only one subgraph that contains every query for all the various underlying protocols, as adding multiple different ones using Apollo can create added complexity. Thus we made this PR.
> 
> I think from a UI integrations perspective @pimato is right about that. Also we will submit a PR to the omen-exchange repo soon that will have a functioning integration based on this commit of the subgraph.
> 
> @pimato your take?

yup, you explained it better than I could.

**Pull request merged**